sphinx/builders/linkcheck.py

```diff
@@ -256,7 +256,8 @@ def check() -> Tuple[str, str, int]:
             uri, docname, lineno = self.wqueue.get()
             if uri is None:
                 break
-            status, info, code = check()
+            srcdir = path.dirname(self.env.doc2path(docname))
+            status, info, code = check(srcdir)
```

Comment:
It would be better to pass docname instead of srcdir.

Reply:
Done: 6b3d4458796d3ed2f536c09b60c2573a276db417